Sources/TuistKit/Graph/GraphManifestLoader.swift

```diff
@@ -138,23 +128,41 @@ class GraphManifestLoader: GraphManifestLoading {
         })
     }
 
+    func loadProject(at path: AbsolutePath) throws -> ProjectDescription.Project {
+        return try loadManifest(.project, at: path)
+    }
+    
+    func loadWorkspace(at path: AbsolutePath) throws -> ProjectDescription.Workspace {
+        return try loadManifest(.workspace, at: path)
+    }
+    
     func loadSetup(at path: AbsolutePath) throws -> [Upping] {
-        let setupPath = path.appending(component: "Setup.swift")
+        let setupPath = path.appending(component: Manifest.setup.fileName)
         guard fileHandler.exists(setupPath) else {
-            throw GraphManifestLoaderError.setupNotFound(path)
+            throw GraphManifestLoaderError.manifestNotFound(.setup, path)
         }
 
-        let setup = try loadManifest(path: setupPath)
-        let actionsJson: [JSON] = try setup.get("actions")
+        let setup = try loadManifestData(at: setupPath)
+        let setupJson = try JSON(data: setup)
+        let actionsJson: [JSON] = try setupJson.get("actions")
         return try actionsJson.compactMap {
             try Up.with(dictionary: $0,
                         projectPath: path,
                         fileHandler: fileHandler) }
     }
 
-    // MARK: - Fileprivate
+    // MARK: - Private
 
-    fileprivate func loadManifest(path: AbsolutePath) throws -> JSON {
+    private func loadManifest<T: Decodable>(_ manifest: Manifest, at path: AbsolutePath) throws -> T {
+        let manifestPath = path.appending(component: manifest.fileName)
+        guard fileHandler.exists(manifestPath) else {
+            throw GraphManifestLoaderError.manifestNotFound(manifest, path)
+        }
+        let data = try loadManifestData(at: manifestPath)
+        return try JSONDecoder().decode(T.self, from: data)
```

Comment:
Not a big deal but considering that we use the manifest loader when we load each project of the graph we could make the decoder an attribute of the manifest loader class.